src/Acuminator/Acuminator.Utils/RoslynExtensions/PXFieldAttributes/AttributeInformation.cs

```diff
@@ -0,0 +1,155 @@
+			{
+				var allAttributes = attributeSymbol.GetAllAttributesDefinedOnThisAndBaseTypes();
+				foreach (var attribute in allAttributes)
+				{
+					if (!attribute.GetBaseTypes().Contains(_context.AttributeTypes.PXEventSubscriberAttribute)) 
+						continue;
+
+					results.Add(attribute);
+					VisitAggregateAttribute(attribute, 10);
+				}
+			}
+			return results;
+
+			void VisitAggregateAttribute(ITypeSymbol _attributeSymbol, int depth)
+			{
+				if (depth < 0)
+					return;
+
+				if (expand)
+				{
+					foreach (var type in _attributeSymbol.GetBaseTypesAndThis())
+					{
+						if (!type.GetBaseTypes().Contains(_context.AttributeTypes.PXEventSubscriberAttribute))
+							break;
+
+						results.Add(type);
+					}
+				}
+
+				if (_attributeSymbol.InheritsFromOrEquals(aggregateAttribute) || _attributeSymbol.InheritsFromOrEquals(dynamicAggregateAttribute))
+				{
+					var allAttributes = _attributeSymbol.GetAllAttributesDefinedOnThisAndBaseTypes();
+					foreach (var attribute in allAttributes)
+					{
+						if (!attribute.GetBaseTypes().Contains(_context.AttributeTypes.PXEventSubscriberAttribute))
+							continue;
+
+						results.Add(attribute);
+						VisitAggregateAttribute(attribute, depth - 1);
+					}
+				}
+				return;
+			}
+		}
+
+		public bool AttributeDerivedFromClass(ITypeSymbol attributeSymbol, ITypeSymbol type)
+		{
+			if (attributeSymbol.InheritsFromOrEquals(type))
+				return true;
+
+			var aggregateAttribute = _context.AttributeTypes.PXAggregateAttribute;
+			var dynamicAggregateAttribute = _context.AttributeTypes.PXDynamicAggregateAttribute;
+
+			if (attributeSymbol.InheritsFromOrEquals( aggregateAttribute) || attributeSymbol.InheritsFromOrEquals(dynamicAggregateAttribute))
+			{
+				var allAttributes = attributeSymbol.GetAllAttributesDefinedOnThisAndBaseTypes();
+				foreach (var attribute in allAttributes)
+				{
+					if (!attribute.GetBaseTypes().Contains(_context.AttributeTypes.PXEventSubscriberAttribute))
+						continue;
+
+					var result = VisitAggregateAttribute(attribute,10);
+
+					if (result)
+						return result;
+				}
+			}
+			return false;
+
+			bool VisitAggregateAttribute(ITypeSymbol _attributeSymbol,int depth)
+			{
+				if (depth < 0)
+					return false;
+
+				if (_attributeSymbol.InheritsFromOrEquals(type))
+					return true;
+
+				if (_attributeSymbol.InheritsFromOrEquals(aggregateAttribute) || _attributeSymbol.InheritsFromOrEquals(dynamicAggregateAttribute))
+				{
+					var allAttributes = _attributeSymbol.GetAllAttributesDefinedOnThisAndBaseTypes();
+					foreach (var attribute in allAttributes)
+					{
+						if (!attribute.GetBaseTypes().Contains(_context.AttributeTypes.PXEventSubscriberAttribute))
+							continue;
+
+						var result = VisitAggregateAttribute(attribute,depth-1);
+
+						if (result)
+							return result;
+					}
+				}
+				return false;
+			}
+		}
+
+		public bool IsBoundAttribute(ITypeSymbol attributeSymbol)
```

Comment:
It is better to have a register of base DB attributes (`PXDBFieldAttribute`, `PXDBCalcedAttribute`) and iterate through it because it can expand in the future.